src/main/java/io/cdap/plugin/gcp/dataplex/source/config/DataplexBatchSourceConfig.java

```diff
@@ -0,0 +1,301 @@
+      }
+    }
+
+    if (fromDate != null && toDate != null && fromDate.isAfter(toDate) && !fromDate.isEqual(toDate)) {
+      collector.addFailure("'Partition From Date' must be before or equal 'Partition To Date'.", null)
+        .withConfigProperty(BigQuerySourceConfig.NAME_PARTITION_FROM)
+        .withConfigProperty(BigQuerySourceConfig.NAME_PARTITION_TO);
+    }
+  }
+
+  private void configureDataplexException(String dataplexConfigProperty, String dataplexConfigPropType,
+                                          DataplexException e,
+                                          FailureCollector failureCollector) {
+    if (("404").equals(e.getCode())) {
+      failureCollector
+        .addFailure("'" + dataplexConfigProperty + "' could not be found. Please ensure that it exists in " +
+          "Dataplex.", null).withConfigProperty(dataplexConfigPropType);
+    } else {
+      failureCollector.addFailure(e.getCode() + ": " + e.getMessage(), null)
+        .withConfigProperty(dataplexConfigPropType);
+    }
+    failureCollector.getOrThrowException();
+  }
+
+  public void validateTable(FailureCollector collector, String project, String dataset, String tableId) {
+    ConfigUtil.validateConnection(this, false, connection, collector);
+    BigQueryUtil.validateTable(tableId, NAME_ENTITY, collector);
+    if (containsMacro(NAME_LOCATION) || containsMacro(NAME_LAKE) || containsMacro(NAME_ZONE) ||
+      containsMacro(NAME_ENTITY)) {
+      return;
+    }
+    if (canConnect()) {
+      TableDefinition.Type definition = getSourceTableType(project, dataset, tableId, collector);
+      if (definition != null && definition == TableDefinition.Type.VIEW) {
+        collector.addFailure(
+          String.format("'%s' is a 'View' :", tableId),
+          "In order to enable query views, please enable 'Enable Querying Views'");
+      }
+    }
+  }
+
+  /**
+   * Returns true if bigquery table can be connected and schema is not a macro.
+   */
+  public boolean canConnect() {
+    return !containsMacro(NAME_SCHEMA) &&
+      connection != null && connection.canConnect();
+  }
+
+  /**
+   * Reads table definition type from BigQuery
+   *
+   * @return {@link TableDefinition.Type}
+   */
+  public TableDefinition.Type getSourceTableType(String datasetProject, String dataSet,
+                                                 String tableId, FailureCollector collector) {
+    Table sourceTable =
+      BigQueryUtil.getBigQueryTable(datasetProject, dataSet, tableId, getServiceAccount(), isServiceAccountFilePath());
+    return sourceTable != null ? sourceTable.getDefinition().getType() : null;
+  }
+
+  public void setupValidatingInputFormat(PipelineConfigurer pipelineConfigurer, FailureCollector collector,
+                                         Entity entity) {
+    ConfigUtil.validateConnection(this, false, connection, collector);
+    String fileFormat = FileFormat.CSV.toString().toLowerCase();
```

Comment:
why are we hardcoding csv file format here?

Reply:
When we run the task API, we set the output format for it and fetch the data accordingly. Here we set it to CSV and will fetch the data from csv files. Due to some issues with CSV input format, we are planning to use Parquet or avro.